File: attached_assets/Starter_code/Dotnet_starter_code/Insurance.Api/Models/Policy.cs
namespace Insurance.Api.Models;

public class Policy
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string PolicyNumber { get; set; } = default!;
    public string Type { get; set; } = default!; // auto | home | life
    public double Premium { get; set; }
    public double Coverage { get; set; }
    public DateOnly StartDate { get; set; }
    public DateOnly EndDate { get; set; }
    public string Status { get; set; } = "pending"; // active | pending | lapsed

    public string CustomerId { get; set; } = default!;
    public Customer Customer { get; set; } = default!;
    public List<Claim> Claims { get; set; } = new();
}

File: attached_assets/Starter_code/Angular_starter_code/src/app/shared/models/policy.ts
export interface Policy {
  id: string;
  policyNumber: string;
  type: 'auto' | 'home' | 'life';
  customerName: string;
  premium: number;
  coverage: number;
  startDate: string;
  endDate: string;
  status: 'active' | 'pending' | 'lapsed';
}


